src/branch/connections.rs

```diff
@@ -32,6 +33,23 @@ impl BranchConnection<'_> {
     }
 }
 
+pub async fn try_connect(connector: &Connector) -> anyhow::Result<Option<Connection>> {
```

Comment:
Please document what does `Ok(None)` response here mean. That the database does not exists?

If this is the case, I think the naming convention would be `connect_if_branch_exists`. `try_connect` is used for stuff that returns a `Result` (i.e. `into()` vs `try_into()`).